[2024-01-22T22:06:27.954+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T22:06:27.958+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T22:06:27.959+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2024-01-22T22:06:27.965+0000] {taskinstance.py:1382} INFO - Executing <Task(SparkSubmitOperator): test> on 2022-12-22 08:00:00+00:00
[2024-01-22T22:06:27.970+0000] {standard_task_runner.py:57} INFO - Started process 153 to run task
[2024-01-22T22:06:27.973+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'first_sample_dag', 'test', 'scheduled__2022-12-22T08:00:00+00:00', '--job-id', '11', '--raw', '--subdir', 'DAGS_FOLDER/test_dag.py', '--cfg-path', '/tmp/tmpk3g5c6hm']
[2024-01-22T22:06:27.977+0000] {standard_task_runner.py:85} INFO - Job 11: Subtask test
[2024-01-22T22:06:28.024+0000] {task_command.py:416} INFO - Running <TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [running]> on host 39178c809964
[2024-01-22T22:06:28.079+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='first_sample_dag' AIRFLOW_CTX_TASK_ID='test' AIRFLOW_CTX_EXECUTION_DATE='2022-12-22T08:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2022-12-22T08:00:00+00:00'
[2024-01-22T22:06:28.087+0000] {spark_submit.py:236} INFO - Could not load connection string spark_default, defaulting to yarn
[2024-01-22T22:06:28.088+0000] {spark_submit.py:366} INFO - Spark-Submit cmd: spark-submit --master yarn --name arrow-spark /usr/local/spark/app/spark.py
[2024-01-22T22:06:28.110+0000] {spark_submit.py:536} INFO - /opt/spark/bin/spark-class: line 71: /usr/lib/jvm/java-8-openjdk-amd64//bin/java: No such file or directory
[2024-01-22T22:06:28.111+0000] {spark_submit.py:536} INFO - /opt/spark/bin/spark-class: line 96: CMD: bad array subscript
[2024-01-22T22:06:28.123+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 467, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --name arrow-spark /usr/local/spark/app/spark.py. Error code is: 1.
[2024-01-22T22:06:28.125+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=first_sample_dag, task_id=test, execution_date=20221222T080000, start_date=20240122T220627, end_date=20240122T220628
[2024-01-22T22:06:28.134+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 11 for task test (Cannot execute: spark-submit --master yarn --name arrow-spark /usr/local/spark/app/spark.py. Error code is: 1.; 153)
[2024-01-22T22:06:28.148+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-01-22T22:06:28.170+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-01-22T22:11:01.538+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T22:11:01.542+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T22:11:01.542+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2024-01-22T22:11:01.548+0000] {taskinstance.py:1382} INFO - Executing <Task(SparkSubmitOperator): test> on 2022-12-22 08:00:00+00:00
[2024-01-22T22:11:01.552+0000] {standard_task_runner.py:57} INFO - Started process 100 to run task
[2024-01-22T22:11:01.555+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'first_sample_dag', 'test', 'scheduled__2022-12-22T08:00:00+00:00', '--job-id', '5', '--raw', '--subdir', 'DAGS_FOLDER/test_dag.py', '--cfg-path', '/tmp/tmp_zdpjwjq']
[2024-01-22T22:11:01.558+0000] {standard_task_runner.py:85} INFO - Job 5: Subtask test
[2024-01-22T22:11:01.588+0000] {task_command.py:416} INFO - Running <TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [running]> on host 19955daf03c7
[2024-01-22T22:11:01.630+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='first_sample_dag' AIRFLOW_CTX_TASK_ID='test' AIRFLOW_CTX_EXECUTION_DATE='2022-12-22T08:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2022-12-22T08:00:00+00:00'
[2024-01-22T22:11:01.637+0000] {spark_submit.py:236} INFO - Could not load connection string spark_default, defaulting to yarn
[2024-01-22T22:11:01.638+0000] {spark_submit.py:366} INFO - Spark-Submit cmd: spark-submit --master yarn --name arrow-spark /usr/local/spark/app/spark.py
[2024-01-22T22:11:01.654+0000] {spark_submit.py:536} INFO - /opt/spark/bin/spark-class: line 71: /usr/lib/jvm/java-8-openjdk-amd64//bin/java: No such file or directory
[2024-01-22T22:11:01.654+0000] {spark_submit.py:536} INFO - /opt/spark/bin/spark-class: line 96: CMD: bad array subscript
[2024-01-22T22:11:01.666+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 467, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --name arrow-spark /usr/local/spark/app/spark.py. Error code is: 1.
[2024-01-22T22:11:01.668+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=first_sample_dag, task_id=test, execution_date=20221222T080000, start_date=20240122T221101, end_date=20240122T221101
[2024-01-22T22:11:01.676+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 5 for task test (Cannot execute: spark-submit --master yarn --name arrow-spark /usr/local/spark/app/spark.py. Error code is: 1.; 100)
[2024-01-22T22:11:01.687+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-01-22T22:11:01.712+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-01-22T22:38:36.298+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T22:38:36.302+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T22:38:36.303+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2024-01-22T22:38:36.309+0000] {taskinstance.py:1382} INFO - Executing <Task(SparkSubmitOperator): test> on 2022-12-22 08:00:00+00:00
[2024-01-22T22:38:36.314+0000] {standard_task_runner.py:57} INFO - Started process 70 to run task
[2024-01-22T22:38:36.317+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'first_sample_dag', 'test', 'scheduled__2022-12-22T08:00:00+00:00', '--job-id', '5', '--raw', '--subdir', 'DAGS_FOLDER/test_dag.py', '--cfg-path', '/tmp/tmpk53mrj9l']
[2024-01-22T22:38:36.320+0000] {standard_task_runner.py:85} INFO - Job 5: Subtask test
[2024-01-22T22:38:36.366+0000] {task_command.py:416} INFO - Running <TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [running]> on host 3441088a5e19
[2024-01-22T22:38:36.425+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='first_sample_dag' AIRFLOW_CTX_TASK_ID='test' AIRFLOW_CTX_EXECUTION_DATE='2022-12-22T08:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2022-12-22T08:00:00+00:00'
[2024-01-22T22:38:36.436+0000] {spark_submit.py:236} INFO - Could not load connection string spark_default, defaulting to yarn
[2024-01-22T22:38:36.436+0000] {spark_submit.py:366} INFO - Spark-Submit cmd: spark-submit --master yarn --name arrow-spark /usr/local/spark/app/spark.py
[2024-01-22T22:38:36.459+0000] {spark_submit.py:536} INFO - /opt/spark/bin/spark-class: line 71: /usr/lib/jvm/java-8-openjdk-amd64//bin/java: No such file or directory
[2024-01-22T22:38:36.460+0000] {spark_submit.py:536} INFO - /opt/spark/bin/spark-class: line 96: CMD: bad array subscript
[2024-01-22T22:38:36.473+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 467, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --name arrow-spark /usr/local/spark/app/spark.py. Error code is: 1.
[2024-01-22T22:38:36.476+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=first_sample_dag, task_id=test, execution_date=20221222T080000, start_date=20240122T223836, end_date=20240122T223836
[2024-01-22T22:38:36.484+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 5 for task test (Cannot execute: spark-submit --master yarn --name arrow-spark /usr/local/spark/app/spark.py. Error code is: 1.; 70)
[2024-01-22T22:38:36.529+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-01-22T22:38:36.546+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-01-22T22:48:09.118+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T22:48:09.126+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T22:48:09.126+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2024-01-22T22:48:09.133+0000] {taskinstance.py:1382} INFO - Executing <Task(SparkSubmitOperator): test> on 2022-12-22 08:00:00+00:00
[2024-01-22T22:48:09.140+0000] {standard_task_runner.py:57} INFO - Started process 95 to run task
[2024-01-22T22:48:09.144+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'first_sample_dag', 'test', 'scheduled__2022-12-22T08:00:00+00:00', '--job-id', '5', '--raw', '--subdir', 'DAGS_FOLDER/test_dag.py', '--cfg-path', '/tmp/tmp5kf7lwae']
[2024-01-22T22:48:09.148+0000] {standard_task_runner.py:85} INFO - Job 5: Subtask test
[2024-01-22T22:48:09.193+0000] {task_command.py:416} INFO - Running <TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [running]> on host 9c1ad29dcd9c
[2024-01-22T22:48:09.261+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='first_sample_dag' AIRFLOW_CTX_TASK_ID='test' AIRFLOW_CTX_EXECUTION_DATE='2022-12-22T08:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2022-12-22T08:00:00+00:00'
[2024-01-22T22:48:09.275+0000] {spark_submit.py:236} INFO - Could not load connection string spark_default, defaulting to yarn
[2024-01-22T22:48:09.276+0000] {spark_submit.py:366} INFO - Spark-Submit cmd: spark-submit --master yarn --name arrow-spark /user/local/spark/app/spark.py
[2024-01-22T22:48:15.190+0000] {spark_submit.py:536} INFO - Exception in thread "main" org.apache.spark.SparkException: When running with master 'yarn' either HADOOP_CONF_DIR or YARN_CONF_DIR must be set in the environment.
[2024-01-22T22:48:15.203+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmitArguments.error(SparkSubmitArguments.scala:634)
[2024-01-22T22:48:15.204+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateSubmitArguments(SparkSubmitArguments.scala:274)
[2024-01-22T22:48:15.204+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateArguments(SparkSubmitArguments.scala:234)
[2024-01-22T22:48:15.205+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmitArguments.<init>(SparkSubmitArguments.scala:119)
[2024-01-22T22:48:15.205+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2$$anon$3.<init>(SparkSubmit.scala:1029)
[2024-01-22T22:48:15.205+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.parseArguments(SparkSubmit.scala:1029)
[2024-01-22T22:48:15.205+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:85)
[2024-01-22T22:48:15.205+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046)
[2024-01-22T22:48:15.206+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055)
[2024-01-22T22:48:15.206+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2024-01-22T22:48:15.289+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 467, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --name arrow-spark /user/local/spark/app/spark.py. Error code is: 1.
[2024-01-22T22:48:15.317+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=first_sample_dag, task_id=test, execution_date=20221222T080000, start_date=20240122T224809, end_date=20240122T224815
[2024-01-22T22:48:15.355+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 5 for task test (Cannot execute: spark-submit --master yarn --name arrow-spark /user/local/spark/app/spark.py. Error code is: 1.; 95)
[2024-01-22T22:48:15.386+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-01-22T22:48:15.506+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-01-22T22:53:28.725+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T22:53:28.730+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T22:53:28.731+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2024-01-22T22:53:28.741+0000] {taskinstance.py:1382} INFO - Executing <Task(SparkSubmitOperator): test> on 2022-12-22 08:00:00+00:00
[2024-01-22T22:53:28.752+0000] {standard_task_runner.py:57} INFO - Started process 77 to run task
[2024-01-22T22:53:28.755+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'first_sample_dag', 'test', 'scheduled__2022-12-22T08:00:00+00:00', '--job-id', '6', '--raw', '--subdir', 'DAGS_FOLDER/test_dag.py', '--cfg-path', '/tmp/tmp49l47upj']
[2024-01-22T22:53:28.758+0000] {standard_task_runner.py:85} INFO - Job 6: Subtask test
[2024-01-22T22:53:28.904+0000] {task_command.py:416} INFO - Running <TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [running]> on host 3b2231e5c840
[2024-01-22T22:53:29.005+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='first_sample_dag' AIRFLOW_CTX_TASK_ID='test' AIRFLOW_CTX_EXECUTION_DATE='2022-12-22T08:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2022-12-22T08:00:00+00:00'
[2024-01-22T22:53:29.018+0000] {spark_submit.py:236} INFO - Could not load connection string spark_default, defaulting to yarn
[2024-01-22T22:53:29.026+0000] {spark_submit.py:366} INFO - Spark-Submit cmd: spark-submit --master yarn --name arrow-spark /user/local/spark/app/spark.py
[2024-01-22T22:53:31.174+0000] {spark_submit.py:536} INFO - Exception in thread "main" org.apache.spark.SparkException: When running with master 'yarn' either HADOOP_CONF_DIR or YARN_CONF_DIR must be set in the environment.
[2024-01-22T22:53:31.176+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmitArguments.error(SparkSubmitArguments.scala:634)
[2024-01-22T22:53:31.177+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateSubmitArguments(SparkSubmitArguments.scala:274)
[2024-01-22T22:53:31.178+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateArguments(SparkSubmitArguments.scala:234)
[2024-01-22T22:53:31.178+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmitArguments.<init>(SparkSubmitArguments.scala:119)
[2024-01-22T22:53:31.179+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2$$anon$3.<init>(SparkSubmit.scala:1029)
[2024-01-22T22:53:31.179+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.parseArguments(SparkSubmit.scala:1029)
[2024-01-22T22:53:31.180+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:85)
[2024-01-22T22:53:31.180+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046)
[2024-01-22T22:53:31.180+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055)
[2024-01-22T22:53:31.181+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2024-01-22T22:53:31.218+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 467, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --name arrow-spark /user/local/spark/app/spark.py. Error code is: 1.
[2024-01-22T22:53:31.223+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=first_sample_dag, task_id=test, execution_date=20221222T080000, start_date=20240122T225328, end_date=20240122T225331
[2024-01-22T22:53:31.238+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 6 for task test (Cannot execute: spark-submit --master yarn --name arrow-spark /user/local/spark/app/spark.py. Error code is: 1.; 77)
[2024-01-22T22:53:31.261+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-01-22T22:53:31.287+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-01-22T23:07:38.046+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T23:07:38.051+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T23:07:38.051+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2024-01-22T23:07:38.059+0000] {taskinstance.py:1382} INFO - Executing <Task(SparkSubmitOperator): test> on 2022-12-22 08:00:00+00:00
[2024-01-22T23:07:38.063+0000] {standard_task_runner.py:57} INFO - Started process 84 to run task
[2024-01-22T23:07:38.066+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'first_sample_dag', 'test', 'scheduled__2022-12-22T08:00:00+00:00', '--job-id', '6', '--raw', '--subdir', 'DAGS_FOLDER/test_dag.py', '--cfg-path', '/tmp/tmpxz4hwb94']
[2024-01-22T23:07:38.069+0000] {standard_task_runner.py:85} INFO - Job 6: Subtask test
[2024-01-22T23:07:38.108+0000] {task_command.py:416} INFO - Running <TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [running]> on host e2c308f391b5
[2024-01-22T23:07:38.184+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='first_sample_dag' AIRFLOW_CTX_TASK_ID='test' AIRFLOW_CTX_EXECUTION_DATE='2022-12-22T08:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2022-12-22T08:00:00+00:00'
[2024-01-22T23:07:38.196+0000] {spark_submit.py:236} INFO - Could not load connection string spark_default, defaulting to yarn
[2024-01-22T23:07:38.198+0000] {spark_submit.py:366} INFO - Spark-Submit cmd: spark-submit --master yarn --conf spark.master=spark://spark:7077 --name arrow-spark /user/local/spark/app/spark.py
[2024-01-22T23:07:39.365+0000] {spark_submit.py:536} INFO - Exception in thread "main" org.apache.spark.SparkException: When running with master 'yarn' either HADOOP_CONF_DIR or YARN_CONF_DIR must be set in the environment.
[2024-01-22T23:07:39.366+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmitArguments.error(SparkSubmitArguments.scala:634)
[2024-01-22T23:07:39.366+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateSubmitArguments(SparkSubmitArguments.scala:274)
[2024-01-22T23:07:39.366+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateArguments(SparkSubmitArguments.scala:234)
[2024-01-22T23:07:39.367+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmitArguments.<init>(SparkSubmitArguments.scala:119)
[2024-01-22T23:07:39.367+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2$$anon$3.<init>(SparkSubmit.scala:1029)
[2024-01-22T23:07:39.367+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.parseArguments(SparkSubmit.scala:1029)
[2024-01-22T23:07:39.367+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:85)
[2024-01-22T23:07:39.368+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046)
[2024-01-22T23:07:39.368+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055)
[2024-01-22T23:07:39.368+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2024-01-22T23:07:39.387+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 467, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --conf spark.master=spark://spark:7077 --name arrow-spark /user/local/spark/app/spark.py. Error code is: 1.
[2024-01-22T23:07:39.391+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=first_sample_dag, task_id=test, execution_date=20221222T080000, start_date=20240122T230738, end_date=20240122T230739
[2024-01-22T23:07:39.401+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 6 for task test (Cannot execute: spark-submit --master yarn --conf spark.master=spark://spark:7077 --name arrow-spark /user/local/spark/app/spark.py. Error code is: 1.; 84)
[2024-01-22T23:07:39.448+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-01-22T23:07:39.479+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-01-22T23:12:53.558+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T23:12:53.562+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T23:12:53.562+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2024-01-22T23:12:53.568+0000] {taskinstance.py:1382} INFO - Executing <Task(SparkSubmitOperator): test> on 2022-12-22 08:00:00+00:00
[2024-01-22T23:12:53.572+0000] {standard_task_runner.py:57} INFO - Started process 69 to run task
[2024-01-22T23:12:53.575+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'first_sample_dag', 'test', 'scheduled__2022-12-22T08:00:00+00:00', '--job-id', '6', '--raw', '--subdir', 'DAGS_FOLDER/test_dag.py', '--cfg-path', '/tmp/tmph7bdnwyk']
[2024-01-22T23:12:53.578+0000] {standard_task_runner.py:85} INFO - Job 6: Subtask test
[2024-01-22T23:12:53.611+0000] {task_command.py:416} INFO - Running <TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [running]> on host b93e82ce7774
[2024-01-22T23:12:53.660+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='first_sample_dag' AIRFLOW_CTX_TASK_ID='test' AIRFLOW_CTX_EXECUTION_DATE='2022-12-22T08:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2022-12-22T08:00:00+00:00'
[2024-01-22T23:12:53.667+0000] {spark_submit.py:236} INFO - Could not load connection string spark_default, defaulting to yarn
[2024-01-22T23:12:53.668+0000] {spark_submit.py:366} INFO - Spark-Submit cmd: spark-submit --master yarn --conf spark.master=spark://spark:7077 --name arrow-spark /user/local/spark/app/spark.py
[2024-01-22T23:12:54.482+0000] {spark_submit.py:536} INFO - Exception in thread "main" org.apache.spark.SparkException: When running with master 'yarn' either HADOOP_CONF_DIR or YARN_CONF_DIR must be set in the environment.
[2024-01-22T23:12:54.483+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmitArguments.error(SparkSubmitArguments.scala:634)
[2024-01-22T23:12:54.484+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateSubmitArguments(SparkSubmitArguments.scala:274)
[2024-01-22T23:12:54.484+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateArguments(SparkSubmitArguments.scala:234)
[2024-01-22T23:12:54.484+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmitArguments.<init>(SparkSubmitArguments.scala:119)
[2024-01-22T23:12:54.484+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2$$anon$3.<init>(SparkSubmit.scala:1029)
[2024-01-22T23:12:54.484+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.parseArguments(SparkSubmit.scala:1029)
[2024-01-22T23:12:54.484+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:85)
[2024-01-22T23:12:54.485+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1046)
[2024-01-22T23:12:54.485+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1055)
[2024-01-22T23:12:54.485+0000] {spark_submit.py:536} INFO - at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2024-01-22T23:12:54.513+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 467, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --conf spark.master=spark://spark:7077 --name arrow-spark /user/local/spark/app/spark.py. Error code is: 1.
[2024-01-22T23:12:54.516+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=first_sample_dag, task_id=test, execution_date=20221222T080000, start_date=20240122T231253, end_date=20240122T231254
[2024-01-22T23:12:54.523+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 6 for task test (Cannot execute: spark-submit --master yarn --conf spark.master=spark://spark:7077 --name arrow-spark /user/local/spark/app/spark.py. Error code is: 1.; 69)
[2024-01-22T23:12:54.554+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-01-22T23:12:54.574+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-01-22T23:21:19.650+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T23:21:19.656+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T23:21:19.656+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2024-01-22T23:21:19.664+0000] {taskinstance.py:1382} INFO - Executing <Task(SparkSubmitOperator): test> on 2022-12-22 08:00:00+00:00
[2024-01-22T23:21:19.674+0000] {standard_task_runner.py:57} INFO - Started process 80 to run task
[2024-01-22T23:21:19.682+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'first_sample_dag', 'test', 'scheduled__2022-12-22T08:00:00+00:00', '--job-id', '5', '--raw', '--subdir', 'DAGS_FOLDER/test_dag.py', '--cfg-path', '/tmp/tmpikjp4ivx']
[2024-01-22T23:21:19.687+0000] {standard_task_runner.py:85} INFO - Job 5: Subtask test
[2024-01-22T23:21:19.734+0000] {task_command.py:416} INFO - Running <TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [running]> on host 9de37719a349
[2024-01-22T23:21:19.801+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='first_sample_dag' AIRFLOW_CTX_TASK_ID='test' AIRFLOW_CTX_EXECUTION_DATE='2022-12-22T08:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2022-12-22T08:00:00+00:00'
[2024-01-22T23:21:19.809+0000] {base.py:73} INFO - Using connection ID 'spark_default' for task execution.
[2024-01-22T23:21:19.810+0000] {spark_submit.py:366} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark:7077 --name arrow-spark /user/local/spark/app/spark.py
[2024-01-22T23:21:21.511+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:21 INFO SparkContext: Running Spark version 3.3.0
[2024-01-22T23:21:21.573+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:21 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-01-22T23:21:21.663+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:21 INFO ResourceUtils: ==============================================================
[2024-01-22T23:21:21.663+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:21 INFO ResourceUtils: No custom resources configured for spark.driver.
[2024-01-22T23:21:21.663+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:21 INFO ResourceUtils: ==============================================================
[2024-01-22T23:21:21.664+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:21 INFO SparkContext: Submitted application: NBA
[2024-01-22T23:21:21.690+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:21 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2024-01-22T23:21:21.701+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:21 INFO ResourceProfile: Limiting resource is cpu
[2024-01-22T23:21:21.701+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:21 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2024-01-22T23:21:21.760+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:21 INFO SecurityManager: Changing view acls to: ***
[2024-01-22T23:21:21.760+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:21 INFO SecurityManager: Changing modify acls to: ***
[2024-01-22T23:21:21.761+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:21 INFO SecurityManager: Changing view acls groups to:
[2024-01-22T23:21:21.761+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:21 INFO SecurityManager: Changing modify acls groups to:
[2024-01-22T23:21:21.761+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:21 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(***); groups with view permissions: Set(); users  with modify permissions: Set(***); groups with modify permissions: Set()
[2024-01-22T23:21:21.965+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:21 INFO Utils: Successfully started service 'sparkDriver' on port 42881.
[2024-01-22T23:21:21.990+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:21 INFO SparkEnv: Registering MapOutputTracker
[2024-01-22T23:21:22.030+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO SparkEnv: Registering BlockManagerMaster
[2024-01-22T23:21:22.050+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2024-01-22T23:21:22.050+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2024-01-22T23:21:22.052+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2024-01-22T23:21:22.071+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-0d3c7dc1-ee2d-48c8-8de4-f74c8fac92ab
[2024-01-22T23:21:22.161+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO MemoryStore: MemoryStore started with capacity 366.3 MiB
[2024-01-22T23:21:22.182+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO SparkEnv: Registering OutputCommitCoordinator
[2024-01-22T23:21:22.440+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2024-01-22T23:21:22.526+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2024-01-22T23:21:22.562+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO TransportClientFactory: Successfully created connection to spark-master/192.168.160.6:7077 after 21 ms (0 ms spent in bootstraps)
[2024-01-22T23:21:22.680+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20240122232122-0003
[2024-01-22T23:21:22.687+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 39227.
[2024-01-22T23:21:22.688+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO NettyBlockTransferService: Server created on 9de37719a349:39227
[2024-01-22T23:21:22.689+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2024-01-22T23:21:22.696+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20240122232122-0003/0 on worker-20240122223729-192.168.160.8-35469 (192.168.160.8:35469) with 6 core(s)
[2024-01-22T23:21:22.697+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO StandaloneSchedulerBackend: Granted executor ID app-20240122232122-0003/0 on hostPort 192.168.160.8:35469 with 6 core(s), 1024.0 MiB RAM
[2024-01-22T23:21:22.698+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 9de37719a349, 39227, None)
[2024-01-22T23:21:22.699+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20240122232122-0003/1 on worker-20240122223729-192.168.160.9-37449 (192.168.160.9:37449) with 6 core(s)
[2024-01-22T23:21:22.699+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO StandaloneSchedulerBackend: Granted executor ID app-20240122232122-0003/1 on hostPort 192.168.160.9:37449 with 6 core(s), 1024.0 MiB RAM
[2024-01-22T23:21:22.702+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO BlockManagerMasterEndpoint: Registering block manager 9de37719a349:39227 with 366.3 MiB RAM, BlockManagerId(driver, 9de37719a349, 39227, None)
[2024-01-22T23:21:22.704+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 9de37719a349, 39227, None)
[2024-01-22T23:21:22.705+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 9de37719a349, 39227, None)
[2024-01-22T23:21:22.902+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20240122232122-0003/0 is now RUNNING
[2024-01-22T23:21:22.928+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:22 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20240122232122-0003/1 is now RUNNING
[2024-01-22T23:21:23.013+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:23 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2024-01-22T23:21:23.132+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:23 WARN SparkSession: Cannot use io.delta.sql.DeltaSparkSessionExtension to configure session extensions.
[2024-01-22T23:21:23.133+0000] {spark_submit.py:536} INFO - java.lang.ClassNotFoundException: io.delta.sql.DeltaSparkSessionExtension
[2024-01-22T23:21:23.133+0000] {spark_submit.py:536} INFO - at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
[2024-01-22T23:21:23.133+0000] {spark_submit.py:536} INFO - at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
[2024-01-22T23:21:23.134+0000] {spark_submit.py:536} INFO - at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
[2024-01-22T23:21:23.134+0000] {spark_submit.py:536} INFO - at java.lang.Class.forName0(Native Method)
[2024-01-22T23:21:23.134+0000] {spark_submit.py:536} INFO - at java.lang.Class.forName(Class.java:348)
[2024-01-22T23:21:23.134+0000] {spark_submit.py:536} INFO - at org.apache.spark.util.Utils$.classForName(Utils.scala:218)
[2024-01-22T23:21:23.134+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.SparkSession$.$anonfun$applyExtensions$1(SparkSession.scala:1221)
[2024-01-22T23:21:23.135+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.SparkSession$.$anonfun$applyExtensions$1$adapted(SparkSession.scala:1219)
[2024-01-22T23:21:23.135+0000] {spark_submit.py:536} INFO - at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2024-01-22T23:21:23.135+0000] {spark_submit.py:536} INFO - at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2024-01-22T23:21:23.135+0000] {spark_submit.py:536} INFO - at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2024-01-22T23:21:23.136+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.SparkSession$.org$apache$spark$sql$SparkSession$$applyExtensions(SparkSession.scala:1219)
[2024-01-22T23:21:23.136+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.SparkSession.<init>(SparkSession.scala:106)
[2024-01-22T23:21:23.136+0000] {spark_submit.py:536} INFO - at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2024-01-22T23:21:23.136+0000] {spark_submit.py:536} INFO - at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[2024-01-22T23:21:23.136+0000] {spark_submit.py:536} INFO - at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2024-01-22T23:21:23.137+0000] {spark_submit.py:536} INFO - at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[2024-01-22T23:21:23.137+0000] {spark_submit.py:536} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:247)
[2024-01-22T23:21:23.137+0000] {spark_submit.py:536} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
[2024-01-22T23:21:23.137+0000] {spark_submit.py:536} INFO - at py4j.Gateway.invoke(Gateway.java:238)
[2024-01-22T23:21:23.137+0000] {spark_submit.py:536} INFO - at py4j.commands.ConstructorCommand.invokeConstructor(ConstructorCommand.java:80)
[2024-01-22T23:21:23.137+0000] {spark_submit.py:536} INFO - at py4j.commands.ConstructorCommand.execute(ConstructorCommand.java:69)
[2024-01-22T23:21:23.138+0000] {spark_submit.py:536} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-01-22T23:21:23.138+0000] {spark_submit.py:536} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-01-22T23:21:23.138+0000] {spark_submit.py:536} INFO - at java.lang.Thread.run(Thread.java:750)
[2024-01-22T23:21:23.190+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:23 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2024-01-22T23:21:23.197+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:23 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2024-01-22T23:21:24.558+0000] {spark_submit.py:536} INFO - Traceback (most recent call last):
[2024-01-22T23:21:24.559+0000] {spark_submit.py:536} INFO - File "/user/local/spark/app/spark.py", line 143, in <module>
[2024-01-22T23:21:24.559+0000] {spark_submit.py:536} INFO - job.run()
[2024-01-22T23:21:24.559+0000] {spark_submit.py:536} INFO - File "/user/local/spark/app/spark.py", line 37, in run
[2024-01-22T23:21:24.560+0000] {spark_submit.py:536} INFO - self.daily_ingest(f"01-02-2017", f"2017-02-01")
[2024-01-22T23:21:24.560+0000] {spark_submit.py:536} INFO - File "/user/local/spark/app/spark.py", line 68, in daily_ingest
[2024-01-22T23:21:24.560+0000] {spark_submit.py:536} INFO - self.spark.sql(f"CREATE SCHEMA IF NOT EXISTS nba_raw")
[2024-01-22T23:21:24.560+0000] {spark_submit.py:536} INFO - File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/session.py", line 1034, in sql
[2024-01-22T23:21:24.562+0000] {spark_submit.py:536} INFO - File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/java_gateway.py", line 1321, in __call__
[2024-01-22T23:21:24.562+0000] {spark_submit.py:536} INFO - File "/opt/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 190, in deco
[2024-01-22T23:21:24.562+0000] {spark_submit.py:536} INFO - File "/opt/spark/python/lib/py4j-0.10.9.5-src.zip/py4j/protocol.py", line 326, in get_return_value
[2024-01-22T23:21:24.604+0000] {spark_submit.py:536} INFO - py4j.protocol.Py4JJavaError: An error occurred while calling o38.sql.
[2024-01-22T23:21:24.604+0000] {spark_submit.py:536} INFO - : org.apache.spark.SparkException: Cannot find catalog plugin class for catalog 'spark_catalog': org.apache.spark.sql.delta.catalog.DeltaCatalog
[2024-01-22T23:21:24.604+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.errors.QueryExecutionErrors$.catalogPluginClassNotFoundForCatalogError(QueryExecutionErrors.scala:1620)
[2024-01-22T23:21:24.605+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.connector.catalog.Catalogs$.load(Catalogs.scala:65)
[2024-01-22T23:21:24.605+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.connector.catalog.CatalogManager.loadV2SessionCatalog(CatalogManager.scala:67)
[2024-01-22T23:21:24.605+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.connector.catalog.CatalogManager.$anonfun$v2SessionCatalog$2(CatalogManager.scala:86)
[2024-01-22T23:21:24.605+0000] {spark_submit.py:536} INFO - at scala.collection.mutable.HashMap.getOrElseUpdate(HashMap.scala:86)
[2024-01-22T23:21:24.606+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.connector.catalog.CatalogManager.$anonfun$v2SessionCatalog$1(CatalogManager.scala:86)
[2024-01-22T23:21:24.606+0000] {spark_submit.py:536} INFO - at scala.Option.map(Option.scala:230)
[2024-01-22T23:21:24.606+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.connector.catalog.CatalogManager.v2SessionCatalog(CatalogManager.scala:85)
[2024-01-22T23:21:24.606+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.connector.catalog.CatalogManager.catalog(CatalogManager.scala:51)
[2024-01-22T23:21:24.606+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.connector.catalog.CatalogManager.currentCatalog(CatalogManager.scala:122)
[2024-01-22T23:21:24.606+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.connector.catalog.LookupCatalog.currentCatalog(LookupCatalog.scala:34)
[2024-01-22T23:21:24.607+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.connector.catalog.LookupCatalog.currentCatalog$(LookupCatalog.scala:34)
[2024-01-22T23:21:24.607+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.analysis.ResolveCatalogs.currentCatalog(ResolveCatalogs.scala:28)
[2024-01-22T23:21:24.607+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.connector.catalog.LookupCatalog$CatalogAndNamespace$.unapply(LookupCatalog.scala:89)
[2024-01-22T23:21:24.607+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.analysis.ResolveCatalogs$$anonfun$apply$1.applyOrElse(ResolveCatalogs.scala:33)
[2024-01-22T23:21:24.607+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.analysis.ResolveCatalogs$$anonfun$apply$1.applyOrElse(ResolveCatalogs.scala:32)
[2024-01-22T23:21:24.607+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.$anonfun$resolveOperatorsDownWithPruning$2(AnalysisHelper.scala:170)
[2024-01-22T23:21:24.607+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:176)
[2024-01-22T23:21:24.608+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.$anonfun$resolveOperatorsDownWithPruning$1(AnalysisHelper.scala:170)
[2024-01-22T23:21:24.608+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.allowInvokingTransformsInAnalyzer(AnalysisHelper.scala:323)
[2024-01-22T23:21:24.608+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperatorsDownWithPruning(AnalysisHelper.scala:168)
[2024-01-22T23:21:24.608+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperatorsDownWithPruning$(AnalysisHelper.scala:164)
[2024-01-22T23:21:24.608+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.resolveOperatorsDownWithPruning(LogicalPlan.scala:30)
[2024-01-22T23:21:24.609+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.$anonfun$resolveOperatorsDownWithPruning$4(AnalysisHelper.scala:175)
[2024-01-22T23:21:24.609+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.trees.UnaryLike.mapChildren(TreeNode.scala:1228)
[2024-01-22T23:21:24.609+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.trees.UnaryLike.mapChildren$(TreeNode.scala:1227)
[2024-01-22T23:21:24.609+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.CreateNamespace.mapChildren(v2Commands.scala:409)
[2024-01-22T23:21:24.609+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.$anonfun$resolveOperatorsDownWithPruning$1(AnalysisHelper.scala:175)
[2024-01-22T23:21:24.610+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.allowInvokingTransformsInAnalyzer(AnalysisHelper.scala:323)
[2024-01-22T23:21:24.610+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperatorsDownWithPruning(AnalysisHelper.scala:168)
[2024-01-22T23:21:24.610+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperatorsDownWithPruning$(AnalysisHelper.scala:164)
[2024-01-22T23:21:24.610+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.resolveOperatorsDownWithPruning(LogicalPlan.scala:30)
[2024-01-22T23:21:24.610+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperatorsWithPruning(AnalysisHelper.scala:99)
[2024-01-22T23:21:24.611+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperatorsWithPruning$(AnalysisHelper.scala:96)
[2024-01-22T23:21:24.611+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.resolveOperatorsWithPruning(LogicalPlan.scala:30)
[2024-01-22T23:21:24.611+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperators(AnalysisHelper.scala:76)
[2024-01-22T23:21:24.611+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper.resolveOperators$(AnalysisHelper.scala:75)
[2024-01-22T23:21:24.611+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.LogicalPlan.resolveOperators(LogicalPlan.scala:30)
[2024-01-22T23:21:24.611+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.analysis.ResolveCatalogs.apply(ResolveCatalogs.scala:32)
[2024-01-22T23:21:24.611+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.analysis.ResolveCatalogs.apply(ResolveCatalogs.scala:28)
[2024-01-22T23:21:24.612+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$2(RuleExecutor.scala:211)
[2024-01-22T23:21:24.612+0000] {spark_submit.py:536} INFO - at scala.collection.LinearSeqOptimized.foldLeft(LinearSeqOptimized.scala:126)
[2024-01-22T23:21:24.612+0000] {spark_submit.py:536} INFO - at scala.collection.LinearSeqOptimized.foldLeft$(LinearSeqOptimized.scala:122)
[2024-01-22T23:21:24.612+0000] {spark_submit.py:536} INFO - at scala.collection.immutable.List.foldLeft(List.scala:91)
[2024-01-22T23:21:24.612+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$1(RuleExecutor.scala:208)
[2024-01-22T23:21:24.612+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$1$adapted(RuleExecutor.scala:200)
[2024-01-22T23:21:24.612+0000] {spark_submit.py:536} INFO - at scala.collection.immutable.List.foreach(List.scala:431)
[2024-01-22T23:21:24.613+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.rules.RuleExecutor.execute(RuleExecutor.scala:200)
[2024-01-22T23:21:24.613+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.analysis.Analyzer.org$apache$spark$sql$catalyst$analysis$Analyzer$$executeSameContext(Analyzer.scala:227)
[2024-01-22T23:21:24.613+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.analysis.Analyzer.$anonfun$execute$1(Analyzer.scala:223)
[2024-01-22T23:21:24.613+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.analysis.AnalysisContext$.withNewAnalysisContext(Analyzer.scala:172)
[2024-01-22T23:21:24.613+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.analysis.Analyzer.execute(Analyzer.scala:223)
[2024-01-22T23:21:24.614+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.analysis.Analyzer.execute(Analyzer.scala:187)
[2024-01-22T23:21:24.614+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$executeAndTrack$1(RuleExecutor.scala:179)
[2024-01-22T23:21:24.614+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.QueryPlanningTracker$.withTracker(QueryPlanningTracker.scala:88)
[2024-01-22T23:21:24.614+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.rules.RuleExecutor.executeAndTrack(RuleExecutor.scala:179)
[2024-01-22T23:21:24.614+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.analysis.Analyzer.$anonfun$executeAndCheck$1(Analyzer.scala:208)
[2024-01-22T23:21:24.614+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.plans.logical.AnalysisHelper$.markInAnalyzer(AnalysisHelper.scala:330)
[2024-01-22T23:21:24.615+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.analysis.Analyzer.executeAndCheck(Analyzer.scala:207)
[2024-01-22T23:21:24.615+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.execution.QueryExecution.$anonfun$analyzed$1(QueryExecution.scala:76)
[2024-01-22T23:21:24.615+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.catalyst.QueryPlanningTracker.measurePhase(QueryPlanningTracker.scala:111)
[2024-01-22T23:21:24.615+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.execution.QueryExecution.$anonfun$executePhase$2(QueryExecution.scala:185)
[2024-01-22T23:21:24.615+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:510)
[2024-01-22T23:21:24.615+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.execution.QueryExecution.$anonfun$executePhase$1(QueryExecution.scala:185)
[2024-01-22T23:21:24.615+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2024-01-22T23:21:24.615+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.execution.QueryExecution.executePhase(QueryExecution.scala:184)
[2024-01-22T23:21:24.616+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.execution.QueryExecution.analyzed$lzycompute(QueryExecution.scala:76)
[2024-01-22T23:21:24.616+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.execution.QueryExecution.analyzed(QueryExecution.scala:74)
[2024-01-22T23:21:24.616+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.execution.QueryExecution.assertAnalyzed(QueryExecution.scala:66)
[2024-01-22T23:21:24.616+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.Dataset$.$anonfun$ofRows$2(Dataset.scala:99)
[2024-01-22T23:21:24.616+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2024-01-22T23:21:24.616+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.Dataset$.ofRows(Dataset.scala:97)
[2024-01-22T23:21:24.616+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.SparkSession.$anonfun$sql$1(SparkSession.scala:622)
[2024-01-22T23:21:24.617+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[2024-01-22T23:21:24.617+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.SparkSession.sql(SparkSession.scala:617)
[2024-01-22T23:21:24.617+0000] {spark_submit.py:536} INFO - at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2024-01-22T23:21:24.617+0000] {spark_submit.py:536} INFO - at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2024-01-22T23:21:24.617+0000] {spark_submit.py:536} INFO - at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-01-22T23:21:24.617+0000] {spark_submit.py:536} INFO - at java.lang.reflect.Method.invoke(Method.java:498)
[2024-01-22T23:21:24.617+0000] {spark_submit.py:536} INFO - at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2024-01-22T23:21:24.618+0000] {spark_submit.py:536} INFO - at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
[2024-01-22T23:21:24.618+0000] {spark_submit.py:536} INFO - at py4j.Gateway.invoke(Gateway.java:282)
[2024-01-22T23:21:24.618+0000] {spark_submit.py:536} INFO - at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2024-01-22T23:21:24.618+0000] {spark_submit.py:536} INFO - at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2024-01-22T23:21:24.618+0000] {spark_submit.py:536} INFO - at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2024-01-22T23:21:24.618+0000] {spark_submit.py:536} INFO - at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2024-01-22T23:21:24.618+0000] {spark_submit.py:536} INFO - at java.lang.Thread.run(Thread.java:750)
[2024-01-22T23:21:24.619+0000] {spark_submit.py:536} INFO - Caused by: java.lang.ClassNotFoundException: org.apache.spark.sql.delta.catalog.DeltaCatalog
[2024-01-22T23:21:24.619+0000] {spark_submit.py:536} INFO - at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
[2024-01-22T23:21:24.619+0000] {spark_submit.py:536} INFO - at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
[2024-01-22T23:21:24.619+0000] {spark_submit.py:536} INFO - at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
[2024-01-22T23:21:24.619+0000] {spark_submit.py:536} INFO - at org.apache.spark.sql.connector.catalog.Catalogs$.load(Catalogs.scala:55)
[2024-01-22T23:21:24.619+0000] {spark_submit.py:536} INFO - ... 85 more
[2024-01-22T23:21:24.620+0000] {spark_submit.py:536} INFO - 
[2024-01-22T23:21:24.656+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:24 INFO SparkContext: Invoking stop() from shutdown hook
[2024-01-22T23:21:24.668+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:24 INFO SparkUI: Stopped Spark web UI at http://9de37719a349:4040
[2024-01-22T23:21:24.671+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:24 INFO StandaloneSchedulerBackend: Shutting down all executors
[2024-01-22T23:21:24.675+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:24 INFO CoarseGrainedSchedulerBackend$DriverEndpoint: Asking each executor to shut down
[2024-01-22T23:21:24.689+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:24 INFO MapOutputTrackerMasterEndpoint: MapOutputTrackerMasterEndpoint stopped!
[2024-01-22T23:21:24.746+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:24 INFO MemoryStore: MemoryStore cleared
[2024-01-22T23:21:24.747+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:24 INFO BlockManager: BlockManager stopped
[2024-01-22T23:21:24.758+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:24 INFO BlockManagerMaster: BlockManagerMaster stopped
[2024-01-22T23:21:24.760+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:24 INFO OutputCommitCoordinator$OutputCommitCoordinatorEndpoint: OutputCommitCoordinator stopped!
[2024-01-22T23:21:24.795+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:24 INFO SparkContext: Successfully stopped SparkContext
[2024-01-22T23:21:24.796+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:24 INFO ShutdownHookManager: Shutdown hook called
[2024-01-22T23:21:24.796+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:24 INFO ShutdownHookManager: Deleting directory /tmp/spark-cf909552-dff7-42cc-a12e-490fa36db45d
[2024-01-22T23:21:24.800+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:24 INFO ShutdownHookManager: Deleting directory /tmp/spark-b0e4fae9-b99a-4268-947d-26fb60210b49/pyspark-e8ca2970-0f35-44f1-b2ec-6acf96afa9b5
[2024-01-22T23:21:24.806+0000] {spark_submit.py:536} INFO - 24/01/22 23:21:24 INFO ShutdownHookManager: Deleting directory /tmp/spark-b0e4fae9-b99a-4268-947d-26fb60210b49
[2024-01-22T23:21:24.888+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 467, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark:7077 --name arrow-spark /user/local/spark/app/spark.py. Error code is: 1.
[2024-01-22T23:21:24.893+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=first_sample_dag, task_id=test, execution_date=20221222T080000, start_date=20240122T232119, end_date=20240122T232124
[2024-01-22T23:21:24.904+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 5 for task test (Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark:7077 --name arrow-spark /user/local/spark/app/spark.py. Error code is: 1.; 80)
[2024-01-22T23:21:24.953+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-01-22T23:21:24.978+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-01-22T23:35:54.124+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T23:35:54.128+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [queued]>
[2024-01-22T23:35:54.128+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2024-01-22T23:35:54.134+0000] {taskinstance.py:1382} INFO - Executing <Task(SparkSubmitOperator): test> on 2022-12-22 08:00:00+00:00
[2024-01-22T23:35:54.140+0000] {standard_task_runner.py:57} INFO - Started process 74 to run task
[2024-01-22T23:35:54.143+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'first_sample_dag', 'test', 'scheduled__2022-12-22T08:00:00+00:00', '--job-id', '5', '--raw', '--subdir', 'DAGS_FOLDER/test_dag.py', '--cfg-path', '/tmp/tmpi_npko__']
[2024-01-22T23:35:54.145+0000] {standard_task_runner.py:85} INFO - Job 5: Subtask test
[2024-01-22T23:35:54.180+0000] {task_command.py:416} INFO - Running <TaskInstance: first_sample_dag.test scheduled__2022-12-22T08:00:00+00:00 [running]> on host 7429c9abb136
[2024-01-22T23:35:54.257+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='first_sample_dag' AIRFLOW_CTX_TASK_ID='test' AIRFLOW_CTX_EXECUTION_DATE='2022-12-22T08:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2022-12-22T08:00:00+00:00'
[2024-01-22T23:35:54.270+0000] {base.py:73} INFO - Using connection ID 'spark_default' for task execution.
[2024-01-22T23:35:54.271+0000] {spark_submit.py:366} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark:7077 --jars /usr/local/spark/resources/delta-core_2.12-2.2.0.jar,/usr/local/spark/resources/spark-avro_2.12-3.3.0.jar,/usr/local/spark/resources/spark-sql-kafka-0-10_2.12-3.3.0.jar --name arrow-spark /user/local/spark/app/spark.py
[2024-01-22T23:35:56.077+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:56 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-01-22T23:35:57.129+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO SparkContext: Running Spark version 3.3.0
[2024-01-22T23:35:57.160+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO ResourceUtils: ==============================================================
[2024-01-22T23:35:57.165+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO ResourceUtils: No custom resources configured for spark.driver.
[2024-01-22T23:35:57.166+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO ResourceUtils: ==============================================================
[2024-01-22T23:35:57.166+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO SparkContext: Submitted application: NBA
[2024-01-22T23:35:57.197+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO ResourceProfile: Default ResourceProfile created, executor resources: Map(cores -> name: cores, amount: 1, script: , vendor: , memory -> name: memory, amount: 1024, script: , vendor: , offHeap -> name: offHeap, amount: 0, script: , vendor: ), task resources: Map(cpus -> name: cpus, amount: 1.0)
[2024-01-22T23:35:57.216+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO ResourceProfile: Limiting resource is cpu
[2024-01-22T23:35:57.217+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO ResourceProfileManager: Added ResourceProfile id: 0
[2024-01-22T23:35:57.363+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO SecurityManager: Changing view acls to: ***
[2024-01-22T23:35:57.364+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO SecurityManager: Changing modify acls to: ***
[2024-01-22T23:35:57.365+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO SecurityManager: Changing view acls groups to:
[2024-01-22T23:35:57.366+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO SecurityManager: Changing modify acls groups to:
[2024-01-22T23:35:57.366+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO SecurityManager: SecurityManager: authentication disabled; ui acls disabled; users  with view permissions: Set(***); groups with view permissions: Set(); users  with modify permissions: Set(***); groups with modify permissions: Set()
[2024-01-22T23:35:57.647+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO Utils: Successfully started service 'sparkDriver' on port 44763.
[2024-01-22T23:35:57.679+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO SparkEnv: Registering MapOutputTracker
[2024-01-22T23:35:57.772+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO SparkEnv: Registering BlockManagerMaster
[2024-01-22T23:35:57.792+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO BlockManagerMasterEndpoint: Using org.apache.spark.storage.DefaultTopologyMapper for getting topology information
[2024-01-22T23:35:57.792+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO BlockManagerMasterEndpoint: BlockManagerMasterEndpoint up
[2024-01-22T23:35:57.795+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO SparkEnv: Registering BlockManagerMasterHeartbeat
[2024-01-22T23:35:57.814+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO DiskBlockManager: Created local directory at /tmp/blockmgr-0191df49-7a9e-4d88-a526-d1b702a700a6
[2024-01-22T23:35:57.826+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO MemoryStore: MemoryStore started with capacity 366.3 MiB
[2024-01-22T23:35:57.836+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:57 INFO SparkEnv: Registering OutputCommitCoordinator
[2024-01-22T23:35:58.066+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO Utils: Successfully started service 'SparkUI' on port 4040.
[2024-01-22T23:35:58.116+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO SparkContext: Added JAR file:///usr/local/spark/resources/delta-core_2.12-2.2.0.jar at spark://7429c9abb136:44763/jars/delta-core_2.12-2.2.0.jar with timestamp 1705966557115
[2024-01-22T23:35:58.117+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO SparkContext: Added JAR file:///usr/local/spark/resources/spark-avro_2.12-3.3.0.jar at spark://7429c9abb136:44763/jars/spark-avro_2.12-3.3.0.jar with timestamp 1705966557115
[2024-01-22T23:35:58.118+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO SparkContext: Added JAR file:///usr/local/spark/resources/spark-sql-kafka-0-10_2.12-3.3.0.jar at spark://7429c9abb136:44763/jars/spark-sql-kafka-0-10_2.12-3.3.0.jar with timestamp 1705966557115
[2024-01-22T23:35:58.217+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO StandaloneAppClient$ClientEndpoint: Connecting to master spark://spark-master:7077...
[2024-01-22T23:35:58.281+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO TransportClientFactory: Successfully created connection to spark-master/192.168.160.6:7077 after 37 ms (0 ms spent in bootstraps)
[2024-01-22T23:35:58.417+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO StandaloneSchedulerBackend: Connected to Spark cluster with app ID app-20240122233558-0002
[2024-01-22T23:35:58.422+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO Utils: Successfully started service 'org.apache.spark.network.netty.NettyBlockTransferService' on port 46833.
[2024-01-22T23:35:58.422+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO NettyBlockTransferService: Server created on 7429c9abb136:46833
[2024-01-22T23:35:58.424+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO BlockManager: Using org.apache.spark.storage.RandomBlockReplicationPolicy for block replication policy
[2024-01-22T23:35:58.431+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO BlockManagerMaster: Registering BlockManager BlockManagerId(driver, 7429c9abb136, 46833, None)
[2024-01-22T23:35:58.434+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO BlockManagerMasterEndpoint: Registering block manager 7429c9abb136:46833 with 366.3 MiB RAM, BlockManagerId(driver, 7429c9abb136, 46833, None)
[2024-01-22T23:35:58.437+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO BlockManagerMaster: Registered BlockManager BlockManagerId(driver, 7429c9abb136, 46833, None)
[2024-01-22T23:35:58.438+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20240122233558-0002/0 on worker-20240122233152-192.168.160.9-35101 (192.168.160.9:35101) with 6 core(s)
[2024-01-22T23:35:58.438+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO BlockManager: Initialized BlockManager: BlockManagerId(driver, 7429c9abb136, 46833, None)
[2024-01-22T23:35:58.439+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO StandaloneSchedulerBackend: Granted executor ID app-20240122233558-0002/0 on hostPort 192.168.160.9:35101 with 6 core(s), 1024.0 MiB RAM
[2024-01-22T23:35:58.450+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO StandaloneAppClient$ClientEndpoint: Executor added: app-20240122233558-0002/1 on worker-20240122233152-192.168.160.7-33605 (192.168.160.7:33605) with 6 core(s)
[2024-01-22T23:35:58.451+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO StandaloneSchedulerBackend: Granted executor ID app-20240122233558-0002/1 on hostPort 192.168.160.7:33605 with 6 core(s), 1024.0 MiB RAM
[2024-01-22T23:35:58.672+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20240122233558-0002/0 is now RUNNING
[2024-01-22T23:35:58.675+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO StandaloneAppClient$ClientEndpoint: Executor updated: app-20240122233558-0002/1 is now RUNNING
[2024-01-22T23:35:58.682+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO StandaloneSchedulerBackend: SchedulerBackend is ready for scheduling beginning after reached minRegisteredResourcesRatio: 0.0
[2024-01-22T23:35:58.839+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO SharedState: Setting hive.metastore.warehouse.dir ('null') to the value of spark.sql.warehouse.dir.
[2024-01-22T23:35:58.845+0000] {spark_submit.py:536} INFO - 24/01/22 23:35:58 INFO SharedState: Warehouse path is 'file:/opt/***/spark-warehouse'.
[2024-01-22T23:36:01.917+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:01 INFO HiveConf: Found configuration file null
[2024-01-22T23:36:01.935+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:01 INFO HiveUtils: Initializing HiveMetastoreConnection version 2.3.9 using Spark classes.
[2024-01-22T23:36:03.126+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:03 INFO HiveClientImpl: Warehouse location for Hive client (version 2.3.9) is file:/opt/***/spark-warehouse
[2024-01-22T23:36:05.393+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:05 INFO CodeGenerator: Code generated in 272.846458 ms
[2024-01-22T23:36:05.541+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:05 INFO CodeGenerator: Code generated in 10.961667 ms
[2024-01-22T23:36:05.764+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:05 INFO SparkContext: Starting job: count at NativeMethodAccessorImpl.java:0
[2024-01-22T23:36:05.787+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:05 INFO DAGScheduler: Registering RDD 2 (count at NativeMethodAccessorImpl.java:0) as input to shuffle 0
[2024-01-22T23:36:05.822+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:05 INFO DAGScheduler: Got job 0 (count at NativeMethodAccessorImpl.java:0) with 1 output partitions
[2024-01-22T23:36:05.823+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:05 INFO DAGScheduler: Final stage: ResultStage 1 (count at NativeMethodAccessorImpl.java:0)
[2024-01-22T23:36:05.826+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:05 INFO DAGScheduler: Parents of final stage: List(ShuffleMapStage 0)
[2024-01-22T23:36:05.828+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:05 INFO DAGScheduler: Missing parents: List()
[2024-01-22T23:36:05.839+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:05 INFO DAGScheduler: Submitting ResultStage 1 (MapPartitionsRDD[5] at count at NativeMethodAccessorImpl.java:0), which has no missing parents
[2024-01-22T23:36:06.037+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:06 INFO MemoryStore: Block broadcast_0 stored as values in memory (estimated size 11.1 KiB, free 366.3 MiB)
[2024-01-22T23:36:06.170+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:06 INFO MemoryStore: Block broadcast_0_piece0 stored as bytes in memory (estimated size 5.4 KiB, free 366.3 MiB)
[2024-01-22T23:36:06.178+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:06 INFO BlockManagerInfo: Added broadcast_0_piece0 in memory on 7429c9abb136:46833 (size: 5.4 KiB, free: 366.3 MiB)
[2024-01-22T23:36:06.182+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:06 INFO SparkContext: Created broadcast 0 from broadcast at DAGScheduler.scala:1513
[2024-01-22T23:36:06.211+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:06 INFO DAGScheduler: Submitting 1 missing tasks from ResultStage 1 (MapPartitionsRDD[5] at count at NativeMethodAccessorImpl.java:0) (first 15 tasks are for partitions Vector(0))
[2024-01-22T23:36:06.213+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:06 INFO TaskSchedulerImpl: Adding task set 1.0 with 1 tasks resource profile 0
[2024-01-22T23:36:09.040+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:09 INFO CoarseGrainedSchedulerBackend$DriverEndpoint: Registered executor NettyRpcEndpointRef(spark-client://Executor) (192.168.160.9:40498) with ID 0,  ResourceProfileId 0
[2024-01-22T23:36:09.042+0000] {spark_submit.py:536} INFO - 24/01/22 23:36:09 INFO CoarseGrainedSchedulerBackend$DriverEndpoint: Registered executor NettyRpcEndpointRef(spark-client://Executor) (192.168.160.7:60008) with ID 1,  ResourceProfileId 0
[2024-01-22T23:36:10.704+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 467, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark:7077 --jars /usr/local/spark/resources/delta-core_2.12-2.2.0.jar,/usr/local/spark/resources/spark-avro_2.12-3.3.0.jar,/usr/local/spark/resources/spark-sql-kafka-0-10_2.12-3.3.0.jar --name arrow-spark /user/local/spark/app/spark.py. Error code is: -9.
[2024-01-22T23:36:10.730+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=first_sample_dag, task_id=test, execution_date=20221222T080000, start_date=20240122T233554, end_date=20240122T233610
[2024-01-22T23:36:10.772+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 5 for task test (Cannot execute: spark-submit --master spark://spark-master:7077 --conf spark.master=spark://spark:7077 --jars /usr/local/spark/resources/delta-core_2.12-2.2.0.jar,/usr/local/spark/resources/spark-avro_2.12-3.3.0.jar,/usr/local/spark/resources/spark-sql-kafka-0-10_2.12-3.3.0.jar --name arrow-spark /user/local/spark/app/spark.py. Error code is: -9.; 74)
[2024-01-22T23:36:10.818+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-01-22T23:36:10.861+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
